pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -258,6 +258,7 @@ func convertToEnvoyJwtConfig(jwtRules []*v1beta1.JWTRule) *envoy_jwt.JwtAuthenti
 			jwtPubKey, err = model.GetJwtKeyResolver().GetPublicKey(jwtRule.JwksUri)
 			if err != nil {
 				log.Errorf("Failed to fetch jwt public key from %q: %s", jwtRule.JwksUri, err)
+				jwtPubKey = `{"keys":[ {"e":"AQAB","kid":"abc","kty":"RSA","n":"failedToFetchJwtPubKey"}]}`
```

Comment:
What is this?

Reply:
This is a fake jwt to resolve the envoy hanging when inline_jwks is empty. The problem is mentioned [here](https://github.com/istio/istio/issues/24629) Thanks.

Reply:
Introducing some magic bad jwt seems problematic. It at least needs a huge comment in the code explaining why we are doing it, and why its safe.

Reply:
Add a comment about why we need this fake jwks (it doesn't block envoy start but would reject all requests with JWT token), and also add a TODO with your github username that this is a short or mid-term fix until we find a better way to handle this case.

Reply:
Why don't we just inject a deny?

Reply:
I think @xulingqing did some investigation but the current Envoy JWT API doesn't provide an way to just deny a request without setting the jwks. And here is using a fake jwks to "workaround" this to achieve the effect of denying a request.

Reply:
Is this a temporary workaround or permanent?

Reply:
a temporary workaround.

@xulingqing can you update the comment to make it clear this is a temporary workaround, and the TODO is to find out a better way to solve this.

Reply:
Sure, working on it

Reply:
`kid` can also be named to something more meaningful rather than `abc`? making debugging life easier.